File: LeetCode/Problems/Python/415. Add Strings.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:

        M = len(num1)
        N = len(num2)
        
        ptr1 = M-1
        ptr2 = N-1
        carry = 0
        result = deque()
        while ptr1 >=0 and ptr2 >=0:
            n1 = ord(num1[ptr1]) - ord('0')
            n2 = ord(num2[ptr2]) - ord('0')
            digit = n1 + n2 + carry
            carry = digit // 10
            digit = digit % 10
            result.appendleft(chr(ord('0')+digit))
            ptr1-=1
            ptr2-=1
        
        
        while ptr1 >=0:
            n1 = ord(num1[ptr1]) - ord('0')
            digit = n1 + carry
            carry = digit // 10
            digit = digit % 10
            result.appendleft(chr(ord('0')+digit))
            ptr1-=1
            
        while ptr2 >=0:
            n2 = ord(num2[ptr2]) - ord('0')
            digit = n2 + carry
            carry = digit // 10
            digit = digit % 10
            result.appendleft(chr(ord('0')+digit))
            ptr2-=1
        
        if carry !=0:
            result.appendleft('1')
        return "".join(result)

File: LeetCode/Problems/C++/415. Add Strings.cpp
class Solution {
public:
    string addStrings(string num1, string num2) {
        if(!num1.size()) return num2;
        if(!num2.size()) return num1;
        
        int n1 = num1.size()-1, n2 = num2.size()-1, sum;
        int carry = 0;
        string result;
        int r = max(n1,n2);
        result.resize(r+1);
        

        while(n1 >=0 && n2 >=0){
            sum = num1[n1] - '0' + num2[n2] - '0'+ carry;
            carry = sum / 10;
            result[r--] = sum % 10 + '0';
            n1--;
            n2--;
        }
        while(n1 >= 0){
            sum = num1[n1] - '0' + carry;
            carry = sum / 10;
            result[r--] = sum %10 + '0';
            n1--;
        }
        while(n2 >=0){
            sum = num2[n2] - '0' + carry;
            carry = sum / 10;
            result[r--] = sum %10 + '0';
            n2--;
        }
        if(carry != 0){
            result = "1" + result;
        }
        return result;
    }
};